src/webui/api/torrentscontroller.cpp

```diff
@@ -661,6 +661,7 @@ void TorrentsController::addAction()
     const bool skipChecking = parseBool(params()[u"skip_checking"_qs]).value_or(false);
     const bool seqDownload = parseBool(params()[u"sequentialDownload"_qs]).value_or(false);
     const bool firstLastPiece = parseBool(params()[u"firstLastPiecePrio"_qs]).value_or(false);
+    const std::optional<bool> addToQueueTop = parseBool(params()[u"addToTopOfQueue"_qs]);
```

Comment:
Just checking: This string is purposefully different than the one in `appcontroller.cpp`, right?
Can you very briefly explain what the code in this function does?

Reply:
> Just checking: This string is purposefully different than the one in `appcontroller.cpp`, right?

Yes.
Names with inconsistent style are used here, but it seems that camelCase style was newer, so I used it.
>Can you very briefly explain what the code in this function does?

Adds new torrent (parse incoming parameters and pass it to corresponding routine).
So all such optional parameters (if provided) are used to override default preferences.